onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.cc

```diff
@@ -51,10 +28,61 @@ void SelectorsAndActions::RegisterSelectorAndAction(const std::string& name,
   ORT_IGNORE_RETURN_VALUE(selectors_and_actions_map_.emplace(name, std::move(entry)));
 }
 
-Status SelectorActionTransformer::MatchAndProcess(Graph& graph, const GraphViewer& graph_viewer,
-                                                  Node& node, bool& modified,
-                                                  const logging::Logger& logger) const {
+#else  // !defined(ORT_MINIMAL_BUILD)
+
+void SelectorsAndActions::RegisterAction(const std::string& name,
+                                         std::unique_ptr<Action> action) {
+  ORT_ENFORCE(actions_map_.find(name) == actions_map_.cend(), "Existing registration with name ", name);
+
+  ORT_IGNORE_RETURN_VALUE(actions_map_.emplace(name, std::move(action)));
+}
+
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
+const Action* SelectorsAndActions::LookUpAction(const std::string& name) const {
+#if !defined(ORT_MINIMAL_BUILD)
+
+  if (const auto it = selectors_and_actions_map_.find(name);
+      it != selectors_and_actions_map_.end() && it->second != nullptr) {
+    return &*it->second->action;
+  }
+  return nullptr;
+
+#else  // !defined(ORT_MINIMAL_BUILD)
+
+  if (const auto it = actions_map_.find(name); it != actions_map_.end()) {
+    return &*it->second;
+  }
+  return nullptr;
+
+#endif  // !defined(ORT_MINIMAL_BUILD)
+}
+
+SelectorActionTransformer::SelectorActionTransformer(const std::string& name,
+                                                     SelectorsAndActions&& selectors_and_actions,
+                                                     const SatApplyContextVariant& apply_context)
+    : GraphTransformer{name},
+      selectors_and_actions_{std::move(selectors_and_actions)},
+      apply_context_{apply_context} {
+#if !defined(ORT_MINIMAL_BUILD)
+  // setup a map so we lookup by operator type efficiently
```

Comment:
Should we have a different ctor for a minimal build? We'd only have actions in that case wouldn't we?

Would it make sense to split apart SelectorAndAction so things are a little simpler between the full build and a minimal build?

Reply:
refactored SelectorAndAction and other helper classes, see if it looks simpler now